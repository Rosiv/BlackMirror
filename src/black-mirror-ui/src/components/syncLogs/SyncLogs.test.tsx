import * as React from 'react';
import renderer from 'react-test-renderer';

import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import darkBaseTheme from 'material-ui/styles/baseThemes/darkBaseTheme';
const customMuiTheme = getMuiTheme(darkBaseTheme);
import SyncLogs from './SyncLogs';

// todo: this test should mock API - component should have dependency on 'store'
// and then mocked store should be injected
// currently this component instantiates and calls API itself which is an antipattern
it('renders and is compatible with the snapshot', () => {
    const p = { params: { id: 'xyz' } };

    const tree = renderer
        .create(
        <MuiThemeProvider muiTheme={customMuiTheme}>
            <SyncLogs
                match={p}
            />
        </MuiThemeProvider>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
