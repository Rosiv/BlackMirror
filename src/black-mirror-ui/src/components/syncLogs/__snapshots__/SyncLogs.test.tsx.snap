// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders and is compatible with the snapshot 1`] = `
<div
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0,0,0,0)",
      "backgroundColor": "#303030",
      "borderRadius": 2,
      "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
             0 1px 4px rgba(0, 0, 0, 0.12)",
      "boxSizing": "border-box",
      "color": "rgba(255, 255, 255, 1)",
      "fontFamily": "Roboto, sans-serif",
      "muiPrepared": true,
      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
      "zIndex": 1,
    }
  }
>
  <div
    style={
      Object {
        "paddingBottom": 8,
      }
    }
  >
    <div
      style={
        Object {
          "muiPrepared": true,
          "padding": 16,
          "position": "relative",
        }
      }
    >
      <span
        style={
          Object {
            "color": "rgba(255, 255, 255, 0.87)",
            "display": "block",
            "fontSize": 24,
            "lineHeight": "36px",
            "muiPrepared": true,
          }
        }
      >
        Logs
      </span>
      <span
        style={
          Object {
            "color": "rgba(255, 255, 255, 0.54)",
            "display": "block",
            "fontSize": 14,
            "muiPrepared": true,
          }
        }
      >
        Retrieving sync undefined status
      </span>
    </div>
    <div
      style={
        Object {
          "color": "rgba(255, 255, 255, 1)",
          "fontSize": 14,
          "muiPrepared": true,
          "padding": 16,
        }
      }
    >
      <div>
        <div
          className=""
        >
          <div
            className="sync-status-container"
          >
            <svg
              className="sync-status-icon"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={
                Object {
                  "color": "rgba(255, 255, 255, 1)",
                  "display": "inline-block",
                  "fill": "#f44336",
                  "height": 24,
                  "muiPrepared": true,
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "userSelect": "none",
                  "width": 24,
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M11 18h2v-2h-2v2zm1-16C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm0-14c-2.21 0-4 1.79-4 4h2c0-1.1.9-2 2-2s2 .9 2 2c0 2-3 1.75-3 5h2c0-2.25 3-2.5 3-5 0-2.21-1.79-4-4-4z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
